back-end/index.js

```diff
@@ -8,15 +8,14 @@ const app = express();
 const PORT = 3000;
 
 app.use(express.json());
+app.use(cors({
+    origin: 'http://localhost:5173', // Specify the exact origin of your frontend
+    credentials: true // Enable credentials to allow sending cookies from the frontend
+}));
 
-const corsOptions = {
-    origin: 'http://localhost:5173',
-    credentials: true,
-};
-app.use(cors(corsOptions));
 
 app.post("/create", async (req, res) => {
-    const { username, password } = req.body;
+    const { username, password, email, name } = req.body;
     try {
```

Comment:
As the app scales, it might be great to start putting these REST endpoints into their own .js files for easier readability.